addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -204,6 +204,15 @@ export class LinkPlugin extends Plugin {
     handleSelectionChange(selection) {
         if (!selection.isCollapsed) {
             this.overlay.close();
+        } else if (!selection.inEditable) {
+            const selection = this.document.getSelection();
+            const preventClosing = selection?.anchorNode?.closest?.(
+                "[data-prevent-closing-overlay]"
+            );
+            if (preventClosing?.dataset?.preventClosingOverlay === "true") {
+                return;
+            }
```

Comment:
This does not seems to handle the case where `anchorNode` is a text node.

Reply:
You could do
```js
const preventClosing = !!findNode(closestPath(selection?.anchorNode), (node) => node.dataset?.preventClosingOverlay);
    if (preventClosing) {
           return;
    }
```

Reply:
The selection and anchorNode is always defined there, no need to check.
```js
const preventClosing = Boolean(findNode(closestPath(selection.anchorNode), (node) => node.dataset?.preventClosingOverlay));
if (preventClosing) {
return;
}

```

Reply:
Thanks for the comments! But I didn't get what you meant about the anchornode is a textnode :confounded: 

Here I'm trying to check when the selection is not in the editable but inside of the linkpopover's editing area (which is the `div` element with `data-prevent-closing-overlay` set to true). So I mean that we just need to check if anchorNode in the editing `div`, cause if the selection is outside of editable + not in the editing overlay, we close the popover anyways

Reply:
@cammarosano the selection here is provided by `onSelectionChange` of the selection pluging and the selection will not be undefined nor null

Reply:
Don't you see a line where `const selection = this.document.getSelection();` is written? It shadows the `selection` parameter...

Reply:
Indeed 🙈

Reply:
Yes I had to use `this.document.getSelection();` cause the selection I'd like to check is not in the editable but in the overlay